app/src/main/java/com/example/countryinfo/CountryDetailsViewModel.kt

```diff
@@ -0,0 +1,68 @@
+package com.example.countryinfo
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.apollographql.apollo.exception.ApolloException
+import io.reactivex.Observable
+import io.reactivex.disposables.Disposable
+
+class CountryDetailsViewModel(private val countryApi: ICountriesApi) : ViewModel() {
+
+    private val countryMutableLiveData = MutableLiveData<CountryDetailsViewState>()
+    val pollLiveData: LiveData<CountryDetailsViewState> = countryMutableLiveData
+    private lateinit var disposable: Disposable
+
+    fun countryClicked(id: String) {
+        getCountryById(id)
+    }
+
+    private fun getCountryById(id: String) {
+        val countryObservable = countryApi.getCountryById(id)
+            .flatMap { response ->
+                val countryData = response.data?.country
```

Comment:
Looking into mapping code piece below, it looks like it's better to check the whole chain for nullability:
```suggestion
                val countryDetails = response.data?.country?.get(0)?.fragments?.countryDetails
```
This way we can reduce a verbosity below...

Reply:
Changed.